src/js/video.js

```diff
@@ -96,10 +96,70 @@ function videojs(id, options, ready) {
   }
 
   // Element may have a player attr referring to an already created player instance.
-  // If not, set up a new player and return the instance.
-  return tag.player || Player.players[tag.playerId] || new Player(tag, options, ready);
+  // If so return that otherwise set up a new player below
+  if (tag.player || Player.players[tag.playerId]) {
+    return tag.player || Player.players[tag.playerId];
+  }
+
+  videojs.hooks('presetup').forEach(function(hookFunction) {
```

Comment:
These are the possible implementations with that naming scheme & the pro's/con's. do you see any other ways to implement it that sound better? (maybe I am just not seeing a good implementation?)   
> Also I wanted to note that using `setup` seem a bit weird to me since it will actually be called `postSetup` and not during setup

### Add lifecycle setters, getters, removers, and lists in a loop to the prototype 
* can make things confusing
* will not duplicate code
* will pollute the video.js object with any lifecycle event functions that we need to add
* may be more intuitive for a new user to call `beforeSetup` and `setup` to access things

### use the hooks terminology internally and add lifecycle  setters, getters, removers, and lists to the prototype in a loop
* Makes me think we should just use hooks directly
* will only pollute the video.js object with wrapper functions
* will not duplicate code
* may be more intuitive for the user

### implement lifecycle functions without any looping or hooks terminology
* Every time we need to expose a new event we have to duplicate functionality (adding a setter, getter, remover, and a list to video.js)
* will pollute video.js with wrapper functions
* lots of duplicate code
* may be intuitive for the user